### DISCUSSION TITLE
[Feature] Add support for tooltips in the head cells, also add an title prop

### DISCUSSION THREAD

**Discussion by BennyAlex** [KevinVandy/material-react-table#1184] [Ideas]
Hello, it would be nice if we can show a tooltip when a header gets hovered. I truncate large texts so they should become fully visible by hovering.
The th should also have the column header as html title prop for accessibility reasons.

**Comment by KevinVandy** [comment]
Use a custom header render to add tooltips, or whatever else you want 
https://www.material-react-table.com/docs/guides/data-columns#custom-header-render

**Reply by BennyAlex** [reply]
Not great to create a custom header for every column manually :D

**Reply by KevinVandy** [reply]
Then use the defaultColumn table option. That applies column def options to every column.

What more are you asking for?

**Reply by BennyAlex** [reply]
Ah perfect, I did not know this prop. 
I look at the examples and havent found that option :)
Thank you!

**Reply by KirillRotanov** [reply]
> Use a custom header render to add tooltips, or whatever else you want https://www.material-react-table.com/docs/guides/data-columns#custom-header-render

It doesn't help in my case. I need to show some button on the right side of the header cell.
I'm trying to implement something like:

```
const CustomColumnHeader: FC<ICustomColumnHeaderProps> = ({ column }) => {
  const [isHovered, setIsHovered] = useState(false);

  const handleMouseEnter = () => {
    setIsHovered(true);
  };

  const handleMouseLeave = () => {
    setIsHovered(false);
  };

  return (
    <div onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave} className={styles.headerCell}>
      {column.columnDef.header}
      {isHovered && (
        <div className={styles.columnDeleteButton}>
          <DeleteIcon />
        </div>
      )}
    </div>
  );
};
export default CustomColumnHeader;
```

and use 
`export const renderCustomHeader = (column: MRT_Column<MyType>) => <CustomColumnHeader column={column} />;`

```
{...
 Header: ({ column }) => renderCustomHeader(column),
 muiTableHeadCellProps: ({ column }) => sortableColumnHeader(column),
 enableSorting: true,
 },
```
 
 ```
.headerCell {
  width: 100%;
  height: 100%;
  border: 1px solid red;
}
```
 
 And I only have access to the actual name of the column, not the whole header cell
![Screenshot 2024-07-30 at 15 18 17](https://github.com/user-attachments/assets/67261635-ab74-48a7-a99c-d5f8991d4e06)
![Screenshot 2024-07-30 at 15 21 23](https://github.com/user-attachments/assets/3cfcca53-7a1f-48d2-aa3d-200b306055a8)

what am i doing wrong?
